contracts/Bridge.sol

```diff
@@ -379,11 +379,11 @@ contract Bridge is Pausable, AccessControl, SafeMath {
         uint72 nonceAndID = (uint72(depositNonce) << 8) | uint72(chainID);
         Proposal storage proposal = _proposals[nonceAndID][dataHash];
 
-        require(proposal._status != ProposalStatus.Cancelled, "Proposal already cancelled");
+        require(proposal._status != ProposalStatus.Cancelled, "Proposal already cancelled"); // REVIEW: can cancel non existent proposals. Malicious relayer can frontrun every voteProposal() with cancelProposal() to block all new proposals from being created.
```

Comment:
We require that a certain amount of blocks have passed after submitting a proposal before being able to cancel it. I don't think this should be an issue.

Reply:
Nvm I'm wrong, it is an issue